internal/controllers/clusterclass/clusterclass_controller.go

```diff
@@ -492,3 +514,69 @@ func matchNamespace(ctx context.Context, c client.Client, selector labels.Select
 	}
 	return selector.Matches(labels.Set(ns.GetLabels()))
 }
+
+const (
+	// ttl is the duration for which we keep entries in the cache.
+	ttl = 10 * time.Minute
+
+	// expirationInterval is the interval in which we will remove expired entries
+	// from the cache.
+	expirationInterval = 10 * time.Hour
+)
+
+func newCallCache() *callCache {
+	r := &callCache{
+		Store: kcache.NewTTLStore(func(obj interface{}) (string, error) {
+			// We only add objects of type callCacheEntry to the cache, so it's safe to cast to E.
+			return obj.(callCacheEntry).Key(), nil
+		}, ttl),
+	}
+	go func() {
+		for {
+			// Call list to clear the cache of expired items.
+			// We have to do this periodically as the cache itself only expires
+			// items lazily. If we don't do this the cache grows indefinitely.
+			r.List()
+
+			time.Sleep(expirationInterval)
+		}
+	}()
+	return r
+}
+
+type callCache struct {
+	kcache.Store
+}
+
+// Add adds the given entry to the Cache.
+// Note: entries expire after the ttl.
+func (r *callCache) Add(e callCacheEntry) {
+	// Note: We can ignore the error here because the key func we pass into NewTTLStore never
+	// returns errors.
+	_ = r.Store.Add(e)
+}
+
+// Has checks if the given key (still) exists in the Cache.
+// Note: entries expire after the ttl.
+func (r *callCache) GetByKey(key string) (*runtimehooksv1.DiscoverVariablesResponse, bool) {
+	// Note: We can ignore the error here because GetByKey never returns an error.
+	item, exists, _ := r.Store.GetByKey(key)
+	if exists {
+		return item.(callCacheEntry).Response, true
+	}
+	return &runtimehooksv1.DiscoverVariablesResponse{}, false
```

Comment:
isn't it enough to return nil here? So we don't need the boolean.

Reply:
Code was removed